server/src/main/java/org/apache/druid/server/initialization/ServerConfig.java

```diff
@@ -34,12 +34,45 @@
  */
 public class ServerConfig
 {
-
   public static final int DEFAULT_GZIP_INFLATE_BUFFER_SIZE = 4096;
 
+  public ServerConfig(
```

Comment:
Should this be done as `@JsonCreator` to get rid of the need for the empty constructor or is there something in this that doesn't come from json?

Reply:
The empty constructor is still used by some tests, I added a comment about how the new non-empty constructor is used.

Reply:
I'm surprised that serde works fine with two constructors and without anyone marked `@JsonCreator` ... how does ObjectMapper decide which constructor to use for deserialization ?

Reply:
I'm not sure on the exact details of what ObjectMapper does to deserialize with those constructors, but I added a new `ServerConfigSerdeTest` that checks this behavior